skimage/filters/_gaussian.py

```diff
@@ -113,8 +117,15 @@ def gaussian(image, sigma=1, output=None, mode='nearest', cval=0,
         if len(sigma) != image.ndim:
             sigma = np.concatenate((np.asarray(sigma), [0]))
     image = convert_to_float(image, preserve_range)
-    return ndi.gaussian_filter(image, sigma, mode=mode, cval=cval,
-                               truncate=truncate)
+    if output is not None:
+        if not isinstance(output, np.ndarray):
+            raise ValueError("Provided output argument is not a numpy array")
+        elif image.dtype != output.dtype:
+            msg = ("Invalid output datatype. A new array is assigned.")
+            output = np.empty_like(image)
+            warn(RuntimeWarning(msg))
+    return ndi.gaussian_filter(image, sigma, output=output, mode=mode,
```

Comment:
Looks like with minimum scipy version 0.19 supported by skimage, `ndi.gaussian_filter` doesn't return anything if `output` is not `None`! that's what is causing CI failure... I wasn't completely mistaken in my previous suggestion :-D